### ISSUE TITLE
DataTable: filtering breaks row-state / action-listener call with current var

### ISSUE THREAD

**Issue by tandraschko** [primefaces/primefaces#7367]
Actual:
- filter by "2"
- click button: "doSomething: 0" is printed

Actual:
- filter by "2"
- click button: "doSomething: 2" should be printed

```
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml"
      xmlns:ui="http://xmlns.jcp.org/jsf/facelets"
      xmlns:f="http://xmlns.jcp.org/jsf/core"
      xmlns:p="http://primefaces.org/ui"
      xmlns:h="http://xmlns.jcp.org/jsf/html">

    <h:head>
        <title>PrimeFaces Test</title>
    </h:head>
    <h:body>

        <h:form id="frmTest">
            a#{testView.entries}a
            <p:dataTable value="#{testView.entries}" filteredValue="#{testView.filteredEntries}" var="entry">
                <p:column field="id" />
                <p:column field="name1" />
                <p:column field="name2" />
                <p:column>
                    <p:commandButton process="@this" update="@none" actionListener="#{testView.doSomething(entry)}" value="Action" />
                </p:column>
            </p:dataTable>
            
        </h:form>

    </h:body>
</html>
```

```
package org.primefaces.test;

import java.io.Serializable;
import java.util.Arrays;
import java.util.List;
import java.util.Objects;
import javax.faces.bean.ViewScoped;
import javax.inject.Named;
import lombok.AllArgsConstructor;

import lombok.Data;
import lombok.NoArgsConstructor;

@Named
@ViewScoped
public class TestView implements Serializable {
    
    private List<Entry> filteredEntries;
    
    public List<Entry> getEntries() {
        return Arrays.asList(new Entry(1, "1", "1"), new Entry(2, "2", "2"), new Entry(3, "3", "3"));
    }
    
    public List<Entry> getFilteredEntries() {
        return filteredEntries;
    }

    public void setFilteredEntries(List<Entry> filteredEntries) {
        this.filteredEntries = filteredEntries;
    }

    public void doSomething(Entry entry) {
        System.err.println("doSomething: " + entry.getId());
    }

    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class Entry {
        private int id;
        private String name1;
        private String name2;

        @Override
        public int hashCode() {
            int hash = 7;
            hash = 53 * hash + Objects.hashCode(this.id);
            return hash;
        }

        @Override
        public boolean equals(Object obj) {
            if (this == obj) {
                return true;
            }
            if (obj == null) {
                return false;
            }
            if (getClass() != obj.getClass()) {
                return false;
            }
            final Entry other = (Entry) obj;
            if (!Objects.equals(this.id, other.id)) {
                return false;
            }
            return true;
        }
        
        
    }
}

```

**Comment by tandraschko** [comment]
@melloware @christophs78 could one of you guys create a integration test for it?

**Comment by melloware** [comment]
Yep I think we already have an integration test we can piggy back and add a button to and click it.

**Comment by tandraschko** [comment]
oh, it works fine with "impl Serializable"
must be still something different in my customer app, will come back

**Comment by melloware** [comment]
Yep I have helped a few people on the forums with the exact same issue where they were not using Serializable and adding it to their POJO fixed it.

**Comment by tandraschko** [comment]
added a comment here #7336
i feel my customers problem is somehow different but related

**Comment by tandraschko** [comment]
related and fixed by: #7336

**Comment by potop** [comment]
Can this bug be fixed within 10.0.1 release or smth?
We have just migrated our app to 10.0 and this issue is a blocker for us.
Will need to downgrade to 8.0 for now

**Comment by melloware** [comment]
@potop it has been marked with the "Elite" label so PrimeTek usually reviews those for Elite releases so hopefully it will go in 10.0.4.
